text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,623 @@
+target. We hold tier 1 targets to our highest standard of requirements.
+
+A proposed new tier 1 target must be reviewed and approved by the compiler team
+based on these requirements. In addition, the release team must approve the
+long-term viability of the target, and the additional work of supporting the
+target. For a tier 1 target, this will typically take place via a full RFC
+proposing the target, to be jointly reviewed and approved by the compiler team
+and release team.
+
+In addition, the infrastructure team must approve the integration of the target
+into Continuous Integration (CI), and the tier 1 CI-related requirements. This
+review and approval may take place in a PR adding the target to CI, by an
+infrastructure team member reporting the outcome of a team discussion, or by
+including the infrastructure team in the RFC proposing the target.
+
+- Tier 1 targets must have substantial, widespread interest within the
+  developer community, and must serve the ongoing needs of multiple production
+  users of Rust across multiple organizations or projects. These requirements
+  are subjective, and determined by consensus of the approving teams. A tier 1
+  target may be demoted or removed if it becomes obsolete or no longer meets
+  this requirement.
+- The target must build and pass tests reliably in CI.
+  - The target must not disable an excessive number of tests or pieces of tests
+    in the testsuite in order to do so. This is a subjective requirement.
+- The target must provide as much of the Rust standard library as is feasible
+  and appropriate to provide. For instance, if the target has a standard memory
+  allocator, it must provide an implementation of `alloc` and the associated
+  data structures.
+- Building the target and running the testsuite for the target must not take
+  substantially longer than other targets.
+- If running the testsuite requires additional infrastructure (such as physical
+  systems running the target), the target maintainers must arrange to provide
+  such resources to the Rust project, to the satisfaction and approval of the
+  Rust infrastructure team.
+  - Such resources may be provided via cloud systems, via emulation, or via
+    physical hardware.
+  - If it is not possible to run the target via emulation, these resources must
+    additionally be sufficient for the Rust infrastructure team to make them
+    available for access by Rust team members, for the purposes of development
+    and testing. (Note that the responsibility for doing target-specific
+    development to keep the target well maintained remains with the target
+    maintainers. This requirement ensures that it is possible for other
+    Rust developers to test the target, but does not obligate other Rust
+    developers to make target-specific fixes.)
+- All tier 2 requirements apply.
+
+A tier 1 target may be demoted if it no longer meets these requirements. Any
```

Comment:
Promotion requires an RFC but demotion does not. Is this the correct route to go?

Reply:
I've clarified this. I expect it to require an RFC as well in most cases. I think it's OK to raise the baseline of a tier 1 target (e.g. CPU/OS requirements) with team approvals, without a full RFC.